src/lib/util/logger/Logger.ts

```diff
@@ -0,0 +1,195 @@
+	'componentHandlerError': (ctx: ComponentContext, error: any) => void;
+	'commandRegistered': (command: Command) => void;
+	'commandUnregistered': (command: Command) => void;
+	'componentRegistered': (component: Component) => void;
+	'componentUnregistered': (component: Component) => void;
+	'listenerRegistered': (listener: Listener) => void;
+	'listenerUnregistered': (listener: Listener) => void;
+	'pluginRegistered': (plugin: Plugin) => void;
+}
+
+export const enum LogLevel {
+	TRACE = 1,
+	DEBUG = 2,
+	INFO = 3,
+	TIME = 4,
+	WARN = 5,
+	ERROR = 8,
+	OFF = 99,
+}
+
+export type LogMethods = 'trace' | 'debug' | 'info' | 'time' | 'warn' | 'timeEnd' | 'error';
+
+export declare interface ILogger {
+	on<U extends keyof LoggerEventsInterface>(
+	  event: U, listener: LoggerEventsInterface[U]
+	): this;
+  
+	emit<U extends keyof LoggerEventsInterface>(
+	  event: U, ...args: Parameters<LoggerEventsInterface[U]>
+	): boolean;
+}
+
+export class ILogger extends EventEmitter {
+	TRACE: LogLevel.TRACE;
+	DEBUG: LogLevel.DEBUG;
+	INFO: LogLevel.INFO;
+	TIME: LogLevel.TIME;
+	WARN: LogLevel.WARN;
+	ERROR: LogLevel.ERROR;
+	OFF: LogLevel.OFF;
+	level: LogLevel;
+
+	constructor(level: LogLevel) {
+		super();
+
+		this.level = level;
+
+		this.TRACE = LogLevel.TRACE;
+		this.DEBUG = LogLevel.DEBUG;
+		this.INFO = LogLevel.INFO;
+		this.TIME = LogLevel.TIME;
+		this.WARN = LogLevel.WARN;
+		this.ERROR = LogLevel.ERROR;
+		this.OFF = LogLevel.OFF;
+	}
+
+	public trace(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.TRACE, ...values);
+	}
+
+	public debug(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.DEBUG, ...values);
+	}
+
+	public info(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.INFO, ...values);
+	}
+
+	public time(val: string): void {
+		if (val.length > 0) this.invokeTime(LogLevel.TIME, val, 'start');
+	}
+
+	public timeEnd(val: string): void {
+		if (val.length > 0) this.invokeTime(LogLevel.TIME, val, 'stop');
+	}
+
+	public warn(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.WARN, ...values);
+	}
+
+	public error(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.ERROR, ...values);
+	}
+
+	public invoke(level: LogLevel, ...values: readonly unknown[]): void {
+		if (!this.enabledFor(level)) return;
+
+		let color = '';
+		if (level === LogLevel.TRACE) color = '\x1b[91m';
+		else if (level === LogLevel.DEBUG) color = '\x1b[2m';
+		else if (level === LogLevel.INFO) color = '\x1b[36m';
+		else if (level === LogLevel.WARN) color = '\x1b[93m';
+		else color = '\x1b[91m';
+
+		const method = this.LevelMethods.get(level);
+		const date = new Date();
+
+		// @ts-expect-error Research required ):
+		console[method](`${color}[${Util.pad(date.getHours())}:${Util.pad(date.getMinutes())}:${Util.pad(date.getSeconds())}/${method.toUpperCase()}]\x1b[0m ${values[0]}`, ...values.slice(1));
+	}
+
+	public invokeTime(level: LogLevel, val: string, type: 'start' | 'stop') {
```

Comment:
I dont think we actually use this

Reply:
Yep, but it must keep because minor change.